src/reverse_route.coffee

```diff
@@ -1,11 +1,22 @@
-class ReverseRoute extends String
+class ReverseRoute
 
   constructor: ({@path, @method}) ->
-    super @path
+    @length = @path.length
 
+    # Enable single character access
+    @[i] = character for character, i in @path
 
-  toString: ->
-    @path
+
+  # Forward String methods
+  indexOf: -> @path.indexOf arguments...
+
```

Comment:
Probably a good point, I picked the string prototype methods that I thought made sense that you might want to call on a route. This might be over-engineering though